File: 0012-integer-to-roman/solution.c
char* intToRoman(int num) {
    int numbers[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    const char *rNls[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};

    char* rNl = (char*)malloc(20 * sizeof(char));  
    rNl[0] = '\0';

    for (int i = 0; i < 13; i++) {
        while (num >= numbers[i]) {
            strcat(rNl, rNls[i]);
            num -= numbers[i];
        }
    }
    return rNl; 
}

File: 0012-integer-to-roman/solution.java
class Solution {
    public String intToRoman(int num) {
        StringBuilder sb = new StringBuilder();
        int []nums ={1000,900,500,400,100,90,50,40,10,9,5,4,1};
        String []str ={"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        for(int i = 0 ;i < nums.length;i++){
            if(num ==0 ){
                break;
            }
            while(num>=nums[i]){
                sb.append(str[i]);
                num-=nums[i];            
        }
        }

        return sb.toString();
    }
}
